### PR TITLE
Added unit tests to utils functions

### PR DESCRIPTION
**Pull request by rbadguy**
Added unit testing  for the following functions: 

- SubdomainRegex()
- NewUniqueElements()
- UniqueAppend()
- RemoveAsteriskLabel()

File is located at Amass/amass/utils/misc_test.go.

### CODE REVIEW

File: amass/utils/misc_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package utils
+
+import (
+	"fmt"
```
Comment (fork-while-fork) [inline amass/utils/misc_test.go]: Is anything from `fmt` being used?

File: amass/utils/misc_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+func TestSubdomainRegex(t *testing.T) {
+	tests := []struct {
+		name     string
+		domain   string
+		event    string
+		expected string
+	}{
+
+		{"subdomain", "owasp.org", "subdomain.owasp.org", "subdomain.owasp.org"},
```
Comment (fork-while-fork) [inline amass/utils/misc_test.go]: The name field in these test tables should conform to a standard format for consistency.

File: amass/utils/misc_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+func TestUniqueAppend(t *testing.T) {
+	tests := []struct {
+		name     string
+		orig     []string
+		event    string
+		expected []string
+	}{
+		{"Test 1: Duplicate elements", []string{"sub1.owasp.org", "sub2.owasp.org"}, "sub2.owasp.org", []string{"sub1.owasp.org", "sub2.owasp.org"}},
```
Comment (fork-while-fork) [inline amass/utils/misc_test.go]: For completeness, I think we should have the same tests for NewUniqueElement and UniqueAppend. Add a test like the "empty return" test for NewUniqueElement

File: amass/utils/misc_test.go (hunk 1 above)
Comment (rbadguy) [reply amass/utils/misc_test.go]: Left over from debugging, deleted the import.

File: amass/utils/misc_test.go (hunk 2 above)
Comment (rbadguy) [reply amass/utils/misc_test.go]: Fixed the two tests that did not follow the "Test#: Foo" convention for the Name field.

File: amass/utils/misc_test.go (hunk 3 above)
Comment (rbadguy) [reply amass/utils/misc_test.go]: Added "Test 2: New element" to TestUniqueAppend.
